src/util/macro_util.rs

```diff
@@ -32,6 +32,17 @@ use crate::{
     Immutable, IntoBytes, Ptr, TryFromBytes, Unalign, ValidityError,
 };
 
+/// Projects the type of the field at `Index` in `Self`.
+///
+/// # Safety
+///
+/// Unsafe code may assume that this accurately reflects the definition of
+/// `Self`.
```

Comment:
Document what the `Index` type represents? IMO, "index" is an odd name since this will really end up being a type like `__Zerocopy_Field_a`. I assume this is a holdover from when it was a numerical const index?

Reply:
It's an index in the most general sense — a label used for lookups. When we use `Field` to project into enums (whose fields cannot have visibility modifiers), we'll use a single `#vis struct Index<const N: usize>` type.

Reply:
[Done](https://github.com/google/zerocopy/compare/c7cfc8df03d86f6f93d73c8bc37729ad12eebd96..2cfcc5a8c3baf2a66dbb18fe633ba234a411a155#diff-2abf6454394ca2a0779f4bf7cc0631f23cf1dc484566b5937e8a1ad9a5b9e3e8R37-R39)